src/lint/collect-header-diagnostics.ts

```diff
@@ -0,0 +1,104 @@
+import type { LintingError } from './algorithm-error-reporter-type';
+
+import { getLocation } from './utils';
+
+const ruleId = 'header-format';
+
+export function collectHeaderDiagnostics(
+  dom: any,
+  headers: { element: Element; contents: string }[]
+) {
+  let lintingErrors: LintingError[] = [];
+
+  for (let { element, contents } of headers) {
+    if (!/\(.*\)$/.test(contents) || / Operator \( `[^`]+` \)$/.test(contents)) {
+      continue;
+    }
+
+    function indexToLineAndColumn(index: number) {
+      let headerLines = contents.split('\n');
+      let headerLine = 0;
+      let seen = 0;
+      while (true) {
+        if (seen + headerLines[headerLine].length >= index) {
+          break;
+        }
+        seen += headerLines[headerLine].length + 1; // +1 for the '\n'
+        ++headerLine;
+      }
+      let headerColumn = index - seen;
+
+      let elementLoc = getLocation(dom, element);
+      let line = elementLoc.startTag.line + headerLine;
+      let column =
+        headerLine === 0
+          ? elementLoc.startTag.col +
+            (elementLoc.startTag.endOffset - elementLoc.startTag.startOffset) +
+            headerColumn
+          : headerColumn + 1;
+
+      return { line, column };
+    }
+
+    let name = contents.substring(0, contents.indexOf('('));
+    let params = contents.substring(contents.indexOf('(') + 1, contents.length - 1);
+
+    if (/ $/.test(name)) {
```

Comment:
```suggestion
    if (/[^\s] $/.test(name)) {
```

This way we can get an error about spacing if there's more than one space instead of an error about naming. We can amend the message to say exactly one.